ENIGMAsystem/SHELL/Universal_System/estring.cpp

```diff
@@ -189,12 +189,11 @@ size_t string_pos(string substr,string str) {
   return res == string::npos ? 0 : (int)res;
 }
 
-string string_format(double val, unsigned tot, unsigned dec)
-{
-  char sbuf[19]; sbuf[0] = 0;
-  sprintf(sbuf,"%0*.*f",tot,dec,val);
-  const string fstr = sbuf;
-  return fstr.c_str();
+string string_format(double val, unsigned tot, unsigned dec) {
+  std::vector<char> sbuf(19 + tot + dec);
```

Comment:
can we get some kind of reasoning for this magic number here?

Reply:
Yeah what's up with `19`?

Reply:
This was the number Robert was using initially for the buffer size. I think he copied it from another routine, where the maximum size of a written number was 18 characters. But I don't know; I only kept it because I wanted a padding that would contain any default behavior from users specifying 0 for those widths.